archaius2-core/src/main/java/com/netflix/archaius/ConfigProxyFactory.java

```diff
@@ -293,6 +321,38 @@ public T invoke(Object Obj, Object[] args) {
         return (T) Proxy.newProxyInstance(type.getClassLoader(), new Class[] { type }, handler);
     }
     
+    protected <T> MethodInvoker<T> createCustomProperty(final Function<String, T> converter, final String propName) {
+        final Property<T> prop = propertyFactory
+                .getProperty(propName)
+                .asType(converter, "");
+        return new MethodInvoker<T>() {
+            @Override
+            public T invoke(Object obj, Object[] args) {
+                return prop.get();
+            }
+
+            @Override
+            public String getKey() {
+                return prop.getKey();
+            }
+        };
+    }
+    
+    private MethodInvoker<?> createCollectionProperty(String propName, ParameterizedType type, Supplier<Collection> listSupplier) {
+        final Class<?> valueType = (Class<?>)type.getActualTypeArguments()[0];
+        return createCustomProperty(s -> { 
+            Collection list = listSupplier.get();
+            if (!s.isEmpty()) {
+                Arrays.asList(s.split("\\s*,\\s*")).forEach(v -> {
+                    if (!v.isEmpty() || valueType == String.class) { 
```

Comment:
Strange that we should need this. Decoder should have a method supportsType to determine if the decoder implementation you're using supports the type.